internal/rotate.go

```diff
@@ -104,15 +127,18 @@ func (h *CertRotationHandler) Rotate() error {
 		} else {
 			log.Printf("Executing step: %s", step.Name())
 			if err := step.Execute(h); err != nil {
```

Comment:
This `err` overwrites the top-level `err` variable... is this intentional?

If yes - I advice to declare a separate variable for this use case which name would explain why it is different.

Reply:
ea2d92f6e660149d175107397ce4ecdbbd2b90aa